Packages/com.unity.inputsystem/InputSystem/Plugins/XR/Daydream.cs

```
@@ -0,0 +1,241 @@
+
+        public Vector2Control touchpad { get; private set; }
+        public ButtonControl volumeUp { get; private set; }
+        public ButtonControl recentered { get; private set; }
+        public ButtonControl volumeDown { get; private set; }
+        public ButtonControl recentering { get; private set; }
+        public ButtonControl app { get; private set; }
+        public ButtonControl home { get; private set; }
+        public ButtonControl touchpadClick { get; private set; }
+        public ButtonControl touchpadTouch { get; private set; }
+
+        public IntegerControl trackingState { get; private set; }
+        public ButtonControl isTracked { get; private set; }
+        public Vector3Control devicePosition { get; private set; }
+        public QuaternionControl deviceRotation { get; private set; }
+        public Vector3Control deviceVelocity { get; private set; }
+        public Vector3Control deviceAcceleration { get; private set; }
+
+        protected override void FinishSetup(InputControlSetup setup)
+        {
+            base.FinishSetup(setup);
+
+            try
+            {
+                XRDeviceDescriptor deviceDescriptor = XRDeviceDescriptor.FromJson(description.capabilities);
+
+                switch (deviceDescriptor.deviceRole)
+                {
+                    case EDeviceRole.LeftHanded:
+                        {
+                            InputSystem.SetUsage(this, CommonUsages.LeftHand);
+                            leftHand = this;
+                            break;
+                        }
+                    case EDeviceRole.RightHanded:
+                        {
+                            InputSystem.SetUsage(this, CommonUsages.RightHand);
+                            rightHand = this;
+                            break;
+                        }
+                    default:
+                        break;
+                }
+            }
+            catch (Exception)
+            { }

```

This will need a better solution. I'll take a look at this and SetupWithDescription after merging.

---

Please remove the try/catch block. Exceptions from InputSystem should not be swallowed here.